### PR TITLE
Keep original failed step in presence of around

### PR DESCRIPTION
**Pull request by flash-gordon**
Well, it looks like I fixed this, specs copied from #105

### CODE REVIEW

**Comment by flash-gordon** [comment]
gah, fixed specs

File: lib/dry/transaction/step_failure.rb [hunk 1]
```diff
@@ -6,6 +6,16 @@ class StepFailure
       attr_reader :step
       attr_reader :value
 
+      # @api private
+      def self.call(step, value)
+        if self === value
+          value
+        else
+          yield
+          new(step, value)
```
Comment (GustavoCaso) [inline lib/dry/transaction/step_failure.rb:15]: @flash-gordon Is just to understand better what it is doing this `call` method. Why do we need a new `StepFailure` after publishing `:step_failed`?

Comment (flash-gordon) [reply lib/dry/transaction/step_failure.rb:15]: Because it worked like this:
image

Comment (GustavoCaso) [reply lib/dry/transaction/step_failure.rb:15]: Right. good point

**Comment by timriley** [comment]
@flash-gordon Thanks for tackling this. Your solution is elegant!

I've rearranged the specs for this a little (I want to start grouping them better by "topic"), and now I'll merge it in.

Thanks you also, @Hugo-Hache, for getting the work on this started!

FWIW, I think I've identified another bug related to nested steps, but I'll open up another PR for that, no need to clutter this one up.

**Comment by timriley** [comment]
BTW, there was no other bug. Everything is golden :)

**Comment by timriley** [comment]
And it's out in 0.12.1 🎉

**Comment by GustavoCaso** [comment]
Thanks, @Hugo-Hache for the work on this subject. 
Thanks, @flash-gordon for always providing great solutions.
Thanks, @timriley for the final touches.

I really like this community ❤️

**Pull request merged**